Makefile

```diff
@@ -120,22 +120,25 @@ webapp_deps: go_deps bower_components
 
 dev_appserver_deps: gcloud-app-engine-python gcloud-app-engine-go gcloud-cloud-datastore-emulator
 
-chrome: browser_deps
+chrome:
 	if [[ -z "$$(which google-chrome)" ]]; then \
 		if [[ -z "$$(which chromium)" ]]; then \
 			make apt-get-chromium; \
 		fi; \
 		sudo ln -s "$$(which chromium)" /usr/bin/google-chrome; \
 	fi
 
-firefox: browser_deps bzip2
-	if [[ "$$(which firefox)" == "" ]]; then \
-	  $(WPTD_PATH)webdriver/install.sh $$HOME/browsers; \
-		sudo ln -s $(FIREFOX_PATH) /usr/bin/firefox; \
+firefox:
+	if [[ -z "$$(which firefox)" ]]; then \
+		make firefox_install; \
 	fi
 
-browser_deps: wget java
-	sudo apt-get install -qy --no-install-suggests $$(apt-cache depends firefox-esr chromedriver | grep Depends | sed "s/.*ends:\ //" | tr '\n' ' ')
+firefox_install: firefox_deps bzip2 wget java
```

Comment:
This bzip2 dependency will never be met, so this target will always be built. It depends on the `if [[ -z "$$(which firefox)" ]]` in the firefox rule to not always repeat the work. Given that, wouldn't it be just as clear to always inline these bits into the firefox target?

Reply:
OK, you get to pick the style :)